packages/polaris-core/src/middlewares/snapshot-middleware.ts

```diff
@@ -26,13 +26,16 @@ export class SnapshotMiddleware {
             if (this.isNotPaginatedResolver(result, root)) {
                 return result;
             }
-            currentPage =
-                context.requestHeaders.snapRequest || this.snapshotConfiguration.autoSnapshot
-                    ? await this.calculateCurrentPageInSnapshotProcess(context, result)
-                    : await result.getData(0, await result.totalCount());
-
-            this.logger.debug('Snapshot middleware finished job', context);
-            return currentPage;
+            try {
+                currentPage =
+                    context.requestHeaders.snapRequest || this.snapshotConfiguration.autoSnapshot
+                        ? await this.calculateCurrentPageInSnapshotProcess(context, result)
+                        : await result.getData(0, await result.totalCount());
+                this.logger.debug('Snapshot middleware finished job', context);
+                return currentPage;
+            } catch (e) {
+                context.errors = context.errors ? [e, ...context.errors] : [e];
```

Comment:
why not to simply push "e" to context.errors?

Reply:
if context errors is undefined? I don't want to initialize it with an empty array, just so I could push to it